src/notebooks/phdi_analytics.py

```diff
@@ -0,0 +1,297 @@
+    20191, 20121, 22101, 22150, 22041, 22182, 22043, 20151, 22180,
+    22102, 22311, 20190, 22124, 22046, 22151, 22039, 22066, 20124,
+    22303, 22181, 22308, 22044, 20194, 22307, 22060, 22027, 22185,
+    22035, 22081, 22092, 22082, 22095, 22096, 22103, 22107, 22106,
+    22109, 22108, 22118, 22116, 22120, 22119, 22122, 22121, 22158,
+    22156, 22160, 22159, 22161, 22183, 22184, 22199, 22009, 22037,
+    22036, 22047, 22067, 20122, 20153, 20172, 20193, 20192, 20195,
+    20196, 20511
+]
+
+
+def geocode(row):
+    """
+    Placeholder geocoder for now, real data would come geocoded
+
+    Comment for the future -- it would be helpful, in addition,
+    to geocoding with SmartyStreets, to also pull in data,
+    e.g. socioeconomic data, that can be joined at census geography
+    levels (e.g. https://www.neighborhoodatlas.medicine.wisc.edu/
+    Area Deprivation Index created at Census Block Group Level)
+    """
+    index, lat, lng = row
+    try:
+        census = cg.coordinates(lng, lat)['2020 Census Blocks'][0]
+
+        data = dict(geoid=census['GEOID'],
+                    state=census['STATE'],
+                    county=census['COUNTY'],
+                    tract=census['TRACT'],
+                    block=census['BLOCK'],
+                    lat=lat,
+                    lng=lng)
+
+    except Exception:
+        data = dict(lat=lat, lng=lng)
+
+    return data
+
+
+def generate_point(center: Point, radius: int) -> Point:
+    """
+    This is from
+
+    https://stackoverflow.com/questions/31192451/
+    """
+    radius_in_kilometers = radius * 1e-3
+    random_distance = random.random() * radius_in_kilometers
+    random_bearing = random.random() * 360
+    return geodesic(kilometers=random_distance).destination(center, random_bearing)
+
+
+def write_points(points, filename='synthetic_locations.csv'):
+    """
+    Output a CSV so that you can upload to Google Maps
+
+    This is to just check the synthetic points
+    """
+    output_file = os.path.join(OUTPUTS_DIR, filename)
+    with open(output_file, 'w', encoding='UTF8', newline="") as f:
+        writer = csv.writer(f)
+        writer.writerow(['latitude', 'longitude'])
+        for p in points:
+            writer.writerow([p[0], p[1]])
+
+
+def generate_random_date(start_date, end_date):
+    time_between_dates = end_date - start_date
+    days_between_dates = time_between_dates.days
+    random_number_of_days = random.randrange(days_between_dates)
+    random_date = start_date + dt.timedelta(days=random_number_of_days)
+
+    return random_date
+
+
+def generate_covid_positive(row):
+    if row['is_fully_vax'] == 0:
+        covid_positive = random.choices([0, 1], weights=(0.97, 0.03), k=1)
+        return covid_positive[0]
+    else:
+        covid_positive = random.choices([0, 1], weights=(0.99, 0.01), k=1)
+        return covid_positive[0]
+
+
+def create_dataframe(radius, number_of_points, center):
```

Comment:
In production, this function would need to be broken down quite a bit. For example, the generation of the geocoding data (lines 170-177) should be its own function, and this function shouldn't write to a file without some kind of parameter to control that. It's fine for prototype purposes, though; I just wanted to point it out to help reinforce best practices.